winmd2markdown/Program.cpp

```diff
@@ -508,7 +508,8 @@ void Program::process_method(output& ss, const MethodDef& method, string_view re
     paramNames.push_back(p.Name());
   }
   constexpr auto resultParamName = "result";
-  if (!paramNames.empty() && paramNames[0] == resultParamName) {
+  constexpr auto operationParamName = "operation";
+  if (!paramNames.empty() && (paramNames[0] == resultParamName || paramNames[0] == operationParamName)) {
```

Comment:
I wonder if we instead of relying on param name (which might inadvertently be named one of these special names), we should rely on the actual type of the method, i.e. inspect that it is `IAsyncOperation<T>`

Reply:
It seems that the non-void methods will have the first paramName as `result` (sync), `operation` (async), or `returnValue`. Would it make sense to make the below change instead?
```suggestion
  if (!paramNames.empty() && realName.empty() && returnType != "void") {
```